packages/@aws-cdk/aws-servicecatalog/README.md

```diff
@@ -9,8 +9,262 @@
+Rules are specified as a JSON string.
+
+```ts
+const portfolio = new servicecatalog.Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+});
+
+const product = new servicecatalog.Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+portfolio.addProvisioningRules({
+  product: product,
+  rules: {
+    'Rule1': {
+      'Assertions': [
+        {
+          'Assert': { 'Fn::Contains': [['t2.micro', 't2.small'], { 'Ref': 'InstanceType' }] },
+          'AssertDescription': 'Instance type should be t2.micro or t2.small',
+        },
+      ],
+    },
+  },
+});
+```
+
+### Stack Set Constraints
+
+A stack set constraint allows you to configure product deployment options using AWS CloudFormation StackSets. 
+You can specify multiple accounts and regions for the product launch. End users can manage those accounts and determine where products deploy and the order of deployment.
+
+
+```ts 
+const portfolio = new servicecatalog.Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+});
+
+const product = new servicecatalog.Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+const accounts = ['012345678901', '012345678902', '012345678903'];
+
+const regions = ['us-west-1', 'us-east-1', 'us-west-2', 'us-east-1'];
+
+const adminRole = new iam.Role(stack, 'AdminRole', {
+  assumedBy: new iam.AccountRootPrincipal(),
+});
+
+const executionRole = new iam.Role(stack, 'ExecutionRole', {
+  assumedBy: new iam.AccountRootPrincipal(),
+});
+
+portfolio.addStackSetConstraint({
+  accounts: accounts,
+  adminRole: adminRole,
+  executionRole: executionRole,
+  product: product,
+  regions: regions,
+});
+```
+
+### Resource Updates
+
+Resource update constraints allow or disallow end users to update tags on resources associated with an AWS Service Catalog provisioned product. 
+If tag updating is allowed, then new tags associated with the AWS Service Catalog product or portfolio will be applied to provisioned resources during a provisioned product update.
+
+```ts 
+const portfolio = new servicecatalog.Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+});
+
+const product = new servicecatalog.Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+portfolio.allowTagUpdates({
+  product: product,
+  tagUpdateOnProvisionedProductAllowed: true,
+});
```

Comment:
General question: why are all of these methods defined on `Portfolio`?

Wouldn't it be more natural to have them on `Product` instead?

You could go:

```ts
portfolio.addProduct(product);
product.alllowTagUpdates(true);
```

?

Reply:
Our goal is to have these methods be bidirectional, so that they can be called from either product or portfolio, and we call the same underlying functions.  Also, these are added to portfolio-product associations, not just an individual resource.  We decided to not expose the association as an object to keep end user experience simpler.